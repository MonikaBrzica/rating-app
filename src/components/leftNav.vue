<template>
  <div class="nav-container">
    <NavLink v-bind:data="{id: item.id, src: item.src, text: item.text, link: item.link }"
             v-for="item in nav"
             v-bind:key="item.id"
    />
  </div>
</template>
<script>
import NavLink from '../components/navLink'
export default {
  name: 'LeftNav',
  components: {
    NavLink
  },
  data () {
    return {
      nav: [
        {
          id: 1,
          text: 'Today',
          src: 'view-dashboard.svg',
          link: '/'
        },
        {
          id: 2,
          text: 'Reports',
          src: 'file-document.svg',
          link: '/reports'
        },
        {
          id: 3,
          text: 'Settings',
          src: 'cog.svg',
          link: '/settings'
        }
      ]
    }
  }
}
</script>
<style>
.nav-container {
  background-color: #2D3038;
  width: 100px;
  height: 100%;
}
li:last-of-type {
  position: absolute;
  bottom: 0;
}
</style>
