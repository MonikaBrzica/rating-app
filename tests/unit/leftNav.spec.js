import { shallowMount } from '@vue/test-utils'
import leftNav from '../../src/components/leftNav'

describe('leftNav', () => {
  const wrapper = shallowMount(leftNav)
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
