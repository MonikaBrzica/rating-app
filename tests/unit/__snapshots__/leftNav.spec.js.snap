// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leftNav renders correctly 1`] = `
<div
  class="nav-container"
>
  <img
    alt="logo"
    class="logo"
    src="../assets/images/logo.png"
  />
   
  <navlink-stub
    active-class="active"
    class="active"
    data="[object Object]"
    tag="li"
  />
  <navlink-stub
    active-class="active"
    class="active"
    data="[object Object]"
    tag="li"
  />
   
  <navlink-stub
    data="[object Object]"
  />
   
  <div
    class="set-container"
    style="display: none;"
  >
    <settings-stub />
  </div>
</div>
`;
